.github/copilot-instructions.md

```diff
@@ -0,0 +1,171 @@
+- Use expression-bodied members for simple properties and methods
+- Use collection initializers and object initializers where appropriate
+- Follow the existing `.editorconfig` conventions
+
+```csharp
+// ✅ Preferred - Pattern matching
+if (type.StartsWith("System.Collections.Generic.", StringComparison.Ordinal))
+{
+    return !type.Contains("Enumerator") && !type.Contains("Compar") && !type.Contains("Exception");
+}
+
+// ✅ Preferred - Expression-bodied properties
+bool IsPullRequest => GitHubActions?.IsPullRequest ?? false;
+
+// ✅ Preferred - Modern null checking
+return type?.IndexOf('>') > -1 && type.TrimEnd().EndsWith('>');
+```
+
+### Code Organization
+- Use `this.` qualification for instance members (enforced by `.editorconfig`)
+- Organize using statements with `System` directives first
+- Use PascalCase for constants
+- Follow the established namespace structure: `DendroDocs.Extensions` for extension methods
+
+## Build System
+
+### NUKE Build System
+- **Use `nuke` to build the project** instead of direct `dotnet` commands when possible
+- Available build targets: `Clean`, `Restore`, `Compile`, `UnitTests`, `CodeCoverage`, `Pack`, `Push`
+- Default target is `Push` which runs the full CI pipeline
+
+```bash
+# ✅ Preferred - Use NUKE for building
+./build.sh compile
+./build.sh unittests  
+./build.sh pack
+
+# ✅ Available for development
+dotnet build
+dotnet test
+```
+
+### GitVersion Requirements
+- **Always fetch complete git history** when working with GitVersion for proper versioning
+- Use `fetch-depth: 0` in GitHub Actions checkout to ensure GitVersion has access to all tags and commits
+- For local development, ensure you have the full repository history with `git fetch --unshallow` if needed
+
+```yaml
+# ✅ Required in GitHub Actions workflows
+- uses: actions/checkout@v4
+  with:
+    fetch-depth: 0
+
+# ✅ For local development if repository was shallow cloned
+git fetch --unshallow
+```
+
+### Testing
+- All changes must maintain or improve test coverage
+- Use MSTest framework (`[TestClass]`, `[TestMethod]`, `[DataRow]`)
+- Use Shouldly for assertions (`result.ShouldBe(expected)`)
+- Follow the existing test naming pattern: `MethodName_Scenario_ExpectedBehavior`
+
+```csharp
+[TestMethod]
+public void ExtensionMethod_NullInput_ShouldThrow()
+{
+    // Arrange
+    string? input = null;
+    
+    // Act
+    Action action = () => input.SomeExtension();
+    
+    // Assert
+    action.ShouldThrow<ArgumentNullException>()
+        .ParamName.ShouldBe("input");
+}
+```
+
+## Git Workflow and Commits
+
+### Commit History
+- **Prefer a linear commit history** with only descriptive commits
+- **Avoid "initial plan" or "work in progress" commits** in the final PR
+- Use conventional commit format when possible: `type(scope): description`
+- Each commit should represent a complete, working change
+
+```bash
+# ✅ Good commit messages
+feat: add support for generic type detection in diagrams
```

Comment:
I don't like prefixes in commit message, the whole commit message should be a good functional description of the change